test/unit/timeout.test.ts

```diff
@@ -115,3 +123,197 @@ describe('Timeout', function () {
     });
   });
 });
+
+describe('TimeoutContext', function () {
+  describe('TimeoutContext.create', function () {
+    describe('when timeoutMS is a number', function () {
```

Comment:
nit: would you mind changing all the `describe` blocks with "when" in their descriptions to `context` to match our testing guidelines? If not I can change them in mine if it's a hassle.